zb-db/src/main/java/io/camunda/zeebe/db/impl/DbCompositeKey.java

```diff
@@ -7,12 +7,38 @@
  */
 package io.camunda.zeebe.db.impl;
 
+import io.camunda.zeebe.db.ContainsForeignKeys;
 import io.camunda.zeebe.db.DbKey;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import org.agrona.DirectBuffer;
 import org.agrona.MutableDirectBuffer;
 
-public record DbCompositeKey<FirstKeyType extends DbKey, SecondKeyType extends DbKey>(
-    FirstKeyType first, SecondKeyType second) implements DbKey {
+public final class DbCompositeKey<FirstKeyType extends DbKey, SecondKeyType extends DbKey>
```

Comment:
I reverted this back to a simple class because `containedForeignKeys` is internal state that shouldn't be part of the constructor. If `DbCompositeKey` were to stay a record we would have to allow setting `containedForeignKeys` from the outside. We could still verify that the contents of `containedForeignKeys` are correct but API design wise it doesn't make sense IMO.

Reply:
I am not sure this is correct. You can overwrite the constructor https://www.baeldung.com/java-record-keyword#constructors
```
public DBCompositeKey {
  containedForeignKeys = collectContainedForeignKeys(first, second);
}
``` 

should work. Thinking about it, the constructor you have should also work if `DBCompositeKey` is a record